src/base-type.ts

```diff
@@ -453,7 +453,7 @@ export abstract class BaseObjectLikeTypeImpl<ResultType> extends BaseTypeImpl<Re
      * See {@link IntersectionType} for more information about intersections.
      */
     // istanbul ignore next: using ordinary stub instead of module augmentation to lighten the load on the TypeScript compiler
-    and<Other extends BaseObjectLikeTypeImpl<unknown>>(
+    and<Other extends BaseObjectLikeTypeImpl<any>>(
```

Comment:
This was the only change needed for TS@latest compatibility. Will check this change against some of our internal project to be sure.